src/topmodel.c

```diff
@@ -421,8 +421,7 @@ d_alloc(Qobs,(*nstep));
 //NJF This is dangerous, there is no validation between nstep and
 //number of historgram ordinates, which is used to iterate Q in later steps
 //so this could easily overflow if nstep is smaller than historgram ords
-d_alloc(Q,(*nstep)); //NJF TODO validate that this works correctly
-                     //When used in "stand alone" mode
+d_alloc(Q,(*nstep)); 
```

Comment:
Was this ever validated?  I don't believe I went through that code path in detail throughout the dyamic alloc refactoring.  Might want to leave this comment until we are sure there isn't possible issues here in stand alone mode?

Reply:
I am able to run in stand alone mode and with NGEN.  Output files topmod.out and hyd.out are produced and seem correct. Not sure if further validation is needed.

On a tengential note, to run in stand_alone I am using:
```
#!/bin/bash
# -lm links math library
gcc ./extern/topmodel/topmodel/src/main.c \
        ./extern/topmodel/topmodel/src/bmi_topmodel.c \
        ./extern/topmodel/topmodel/src/topmodel.c -o\
        run_topmodel_bmi -lm

./run_topmodel_bmi # ./Topmodel/topmodel_cat-12.run
```
When I view the info in docs/STAND_ALONE.md, I do not see these instructions anywhere. If this is the expected way to execute in stand_alone, it may be worth adding those instructions there.

If it is not the expected way to execute in stand_alone mode, feel free to let me know!

Reply:
@Ben-Choat - The [install](https://github.com/NOAA-OWP/topmodel/blob/master/INSTALL.md#build-and-run) doc gives instructions on standalone via cmake.  It probably wouldn't be a bad idea to add a link in the STAND_ALONE.md.  Thanks for pointing this out.